spec/features/flights/index_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe 'flights index page' do
+  before :each do
+    @airline = Airline.create!(name: "Southwest")
+    @airline_2 = Airline.create!(name: "United")
+
+    @flight = @airline.flights.create!(number: "123", date: "08/05/2024", departure_city: "Denver", arrival_city: "Reno")
+    @flight_2 = @airline_2.flights.create!(number: "456", date: "08/05/2024", departure_city: "Reno", arrival_city: "Denver")
+    
+    @passenger_1 = Passenger.create!(name: "Joe")
+    @passenger_2 = Passenger.create!(name: "Sally")
+    @passenger_3 = Passenger.create!(name: "Bob")
+    @passenger_4 = Passenger.create!(name: "Jill")
+
+    FlightPassenger.create!(flight_id: @flight.id, passenger_id: @passenger_1.id)
+    FlightPassenger.create!(flight_id: @flight.id, passenger_id: @passenger_2.id)
+    FlightPassenger.create!(flight_id: @flight_2.id, passenger_id: @passenger_3.id)
+    FlightPassenger.create!(flight_id: @flight_2.id, passenger_id: @passenger_4.id)
+
+  end
+
+  it 'can see a list of all flight required flight info US1' do
+    visit flights_path
+
+    within "#flight-info-list" do
+      expect(page).to have_content(@flight.number)
+      expect(page).to have_content(@flight_2.number)
+
+      expect(page).to have_content(@passenger_1.name)
+      expect(page).to have_content(@passenger_2.name)
+      expect(page).to have_content(@passenger_3.name)
+      expect(page).to have_content(@passenger_4.name)
```

Comment:
This does not accurately test that the correct passengers are listed with the appropriate flight though. How can you use a within block for each flight instead of for all the flights to ensure the correct passengers are listed?